waspc/data/Generator/templates/server/src/app.js

```diff
@@ -35,7 +35,8 @@ app.use((err, req, res, next) => {
     return res.status(err.statusCode).json({ message: err.message, data: err.data })
   }
 
-  return next(err)
+  console.error(err);
+  return res.status(500).json({message: "An internal server error occurred."})
```

Comment:
This fixes leaking the internal error since we have no downstream handler for `next(err)` to transform it.

Reply:
Dont we have a default error catcher that transforms any errors that are not shaped as HttpError into 500? I believe that is how Wasp works right now. Also, this is not connected with this PR right?

Reply:
I kinda cheated since I knew it was small and this discussion PR would get some eyes to throw it in... so yes, not strictly related, but found during this one. :D

I don't see any default server error catcher/transformer. We do some reshaping on the client, but as of now, anything that is _not_ `instanceof HttpError` we just let it bubble up and leak details to the client. I can pull this out into a separate PR though.

Reply:
Moved here: https://github.com/wasp-lang/wasp/pull/894